app/models/data_layer.rb

```diff
@@ -44,6 +44,7 @@ def check_is_ready
         self.is_ready = false
       end
     end
+    true
```

Comment:
return true at the end of the algorithm to check the is_ready status, as we don't want to halt any saves.